include/aspect/material_model/utilities.h

```diff
@@ -259,6 +266,88 @@ namespace aspect
                           const bool interpol,
                           const MPI_Comm &comm);
         };
+
+        /**
+        * This class reads in entropy-pressure material table and looks up for material
+        * proporties when entropy and pressure are given
+        */
+        class EntropyReader
+        {
+          public:
+
+            /**
+             * Read the table into material_lookup
+             */
+            void
+            initialize(const MPI_Comm comm,
+                       const std::string data_directory,
+                       const std::string material_file_name);
+
+            /**
+             * Reads the specific heat for a given entropy and pressure from material_lookup
```

Comment:
Same here and below, keep the documentation, but remove the mention of `material_lookup`.